### PR TITLE
[sql] support ilike for postgres string

### PR DESCRIPTION
**Pull request by zhuqi-lucas**
Since calcite already support ilike, we can support ilike for postgres string in feldera.

### CODE REVIEW

**Review by mihaibudiu — CHANGES_REQUESTED** [review]
Thank you for the contribution.
Does this really work?
I expect you need to add a line in ExpressionCompiler.java too, very similar to the handling of LIKE.
Did you find this recipe: https://github.com/feldera/feldera/issues/951?
It would be great to add this function in the docs/sql/strings.md table with "operations on string values" as well.

**Comment by zhuqi-lucas** [comment]
Thank you @mihaibudiu for review, let me look into the recipe.

**Comment by mihaibudiu** [comment]
I think you for most important points from the recipe already.
The only two missing things I pointed out in my review.

**Comment by zhuqi-lucas** [comment]
> Thank you for the contribution. Does this really work? I expect you need to add a line in ExpressionCompiler.java too, very similar to the handling of LIKE. Did you find this recipe: #951? It would be great to add this function in the docs/sql/strings.md table with "operations on string values" as well.


I look into the Calcite source code, we don't need to add new ILIKE to ExpressionCompiler.java, because:
1. ILIKE in calcite also reuse the SqlKind.LIKE, so we will match the LIKE sqlkind too.
2. The calcite use ILIKE as a special case, details code:

https://github.com/apache/calcite/blob/ef1a83f659e8771c65c2541b92d2ef9cc2a05bea/core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java#L1113

https://github.com/apache/calcite/blob/ef1a83f659e8771c65c2541b92d2ef9cc2a05bea/core/src/main/java/org/apache/calcite/sql/fun/SqlLikeOperator.java#L142

3. And the test passed in our unit test.

**Comment by mihaibudiu** [comment]
Great, I would still add a comment in the LIKE switch case indicating that this matches ILIKE as well.
I was surprised the test passed.

**Comment by zhuqi-lucas** [comment]
The docs are missing, i will try to add it, thanks!

**Review by mihaibudiu — APPROVED** [review]

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresStringTests.java
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresStringTests.java:828]: you can combine multiple such tests using the qs function. but let's leave them right now as they are.

**Comment by mihaibudiu** [comment]
I have already approved this PR, if CI passes it will get merged into main.
Thank you for the contribution @zhuqi-lucas

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresStringTests.java
Comment (zhuqi-lucas) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresStringTests.java:828]: Good point, we can refactor all those to qs.

**Comment by zhuqi-lucas** [comment]
Thank you @mihaibudiu for review, i am new to the CI of feldera, does feldera CI will run after maintainers approved?

**Comment by mihaibudiu** [comment]
Yes, there will be 2 rounds of CI: this PR is validated, if it passes, it goes to the merge queue.
Once it reaches the head of the merge queue it is merged if there are no conflicts and even more tests are run.
If any of these fails you will get a notification.

**Comment by zhuqi-lucas** [comment]
Got it!

**Pull request merged**
